### PR TITLE
Case agnostic bulkdata

### PR DESCRIPTION
**Pull request by prescod**
# Changes

A pre-release (?) version of CumulusCI was very strict about requiring the case of sObjects and Fields to be exactly the same as the "canonical" Salesforce name. This puts it back to being case agnostic.

### CODE REVIEW

**Comment by prescod** [comment]
> @prescod I think we need a solution which updates the mapping with normalized schema names rather than just making sure they pass validation as written. Otherwise checks like this (coming in Scott's person account PR) will fail when the case is different: https://github.com/SFDO-Tooling/CumulusCI/pull/1829/files#diff-81c24e397db90aa3b439d7c1753048a1R199

File: cumulusci/tasks/bulkdata/mapping_parser.py [hunk 1]
```diff
@@ -123,13 +135,14 @@ def _get_permission_type(self, operation: DataOperationType) -> str:
         return "createable"
 
     def _check_object_permission(
-        self, global_describe: Dict, sobject: str, operation: DataOperationType
+        self, global_describe: Mapping, sobject: str, operation: DataOperationType
     ):
+        assert sobject in global_describe
```
Comment (davidmreed) [inline cumulusci/tasks/bulkdata/mapping_parser.py:140]: `global_describe` will not contain `sobject` if the user doesn't have Read permission at the object level.

Comment (prescod) [reply cumulusci/tasks/bulkdata/mapping_parser.py:140]: That case should be caught on line 235 which is new.

https://github.com/SFDO-Tooling/CumulusCI/pull/1919/files/98a4287b36aa7070c85cfb4896c2baf279f999de#diff-0a8fc3e0cff0ad9e099cacf4f6fb0f41R235

Comment (davidmreed) [reply cumulusci/tasks/bulkdata/mapping_parser.py:140]: Ah, got it. Thanks for clarifying.

File: cumulusci/tasks/bulkdata/tests/test_utils.py [hunk 2]
```diff
@@ -46,6 +46,12 @@ def mock_sobject_describe(name: str):
             body=read_mock(name),
             status=200,
         )
+        responses.add(
+            method="GET",
+            url=f"https://example.com/services/data/v48.0/sobjects/{name.lower()}/describe",
+            body=read_mock(name),
+            status=200,
+        )
```
Comment (davisagli) [inline cumulusci/tasks/bulkdata/tests/test_utils.py:54]:
```suggestion
```

This seems to be unnecessary

File: cumulusci/tasks/bulkdata/mapping_parser.py [hunk 3]
```diff
@@ -13,7 +14,18 @@
...
+
+class CaseInsensitiveDict(RequestsCaseInsensitiveDict):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # there's a more efficient way to do this by poking around the
+        # base class internals but this way may be less fragile
+        self._canonical_keys = {key.lower(): key for key in self.keys()}
+
+    def canonical_key(self, name):
+        return self._canonical_keys[name.lower()]
```
[the remark this answers is not in the document]
Comment (prescod) [reply cumulusci/tasks/bulkdata/mapping_parser.py:27]: Sure.

The alternative alluded to in the comment is that Requests internal data-structure has all of the information we need, but its an undocumented internal data-structure.

File: cumulusci/tasks/bulkdata/tests/test_utils.py (hunk 2 above)
Comment (prescod) [reply cumulusci/tasks/bulkdata/tests/test_utils.py:54]: I think it seems unnecessary on Mac but was added for Linux.

Comment (davisagli) [reply cumulusci/tasks/bulkdata/tests/test_utils.py:54]: But how does this end up lowercase on Linux? The describe call is made using the mapping's sf_object after _validate_sobject was called, so it should be canonicalized based on the objects from the global describe.

**Comment by davisagli** [comment]
@davidmreed Does this look all right to you?

**Comment by davidmreed** [comment]
Yeah, this looks fine to me. @prescod Have you run it through the integration test suite?

**Comment by prescod** [comment]
Integration tests pass.

**Pull request merged**